lib/ExpensiMark.js

```diff
@@ -49,6 +49,38 @@ export default class ExpensiMark {
                 replacement: '$1<code>$2</code>$3',
             },
 
+            /**
+             * Apply the hereMention first because the string @here is still a valid mention for the userMention regex.
+             * This ensures that the hereMention is always considered first, even if it is followed by a valid userMention.
+            */
+            {
+                name: 'hereMentions',
+                regex: /(?<![^\s]+)@here\b/g,
+                replacement: match => `<mention-here>${match}</mention-here>`,
+            },
+
+            /**
+             * This regex matches a valid user mention in a string.
+             * A user mention is a string that starts with the '@' symbol and is followed by a valid user's primary login
+             * The username can contain any combination of letters, numbers, periods, underscores, and hyphens
+             *
+             * Note: currently we are only allowing mentions in a format of @+19728974297@expensify.sms and @username@expensify.com
+             * inorder to match mentions in a format of @username we should update the regex to: (?!@here)+((&#x60;|<code>|<pre>)\s*?)?[`.]?@+[\w+]*
+             */
+            {
+                name: 'userMentions',
+                regex: /((&#x60;|<code>|<pre>)\s*?)?[`.]?@+[\w+]*@+\w*[.]+\w*/gm,
+                replacement: (match) => {
+                    const isMatchValid = !(match.startsWith('.') || /(<code>|<pre>|&#x60;)/g.test(match));
+
+                    if (match.endsWith('_') && isMatchValid) {
+                        const mention = match.substring(0, match.length - 1);
+                        return `<mention-user>${mention}</mention-user>_`;
```

Comment:
Can you please add a comment to explain what's going on with the trailing underscore? It doesn't make sense to me from just looking at the code.

Reply:
I think updating the regex makes sense 👍